managed/CounterStrikeSharp.API/Modules/Admin/AdminManager.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+using System.Text.Json;
+using CounterStrikeSharp.API.Core;
+using CounterStrikeSharp.API.Modules.Utils;
+using CounterStrikeSharp.API.Modules.Entities;
+using CounterStrikeSharp.API.Modules.Commands;
+using System.Linq;
+using System.Reflection;
+
+namespace CounterStrikeSharp.API.Modules.Admin
+{
+    public partial class AdminData
+    {
+        [JsonPropertyName("auth_type")] public string AuthType { get; set; }
+        [JsonPropertyName("identity")] public string Identity { get; set; }
+        [JsonPropertyName("flags")] public List<string> Flags { get; set; }
+    }
+
+    public static class AdminManager
+    {
+        private static Dictionary<string, AdminData> _admins = new Dictionary<string, AdminData>();
```

Comment:
We should try and type this as a `SteamID` lookup, then the `GetAdminData` method can just do a simple dictionary lookup. We may need to implement `IEquatable<SteamID>` using the `SteamId64` as the underlying value for this to work though.
```suggestion
        private static Dictionary<SteamID, AdminData> _admins = new Dictionary<SteamID, AdminData>();
```

Reply:
I tried doing this with `IEquatable` but I wasn't able to get it working. I compromised by changing the key type to `ulong` for SteamID64's. I wouldn't be opposed if you wanted to take a crack at implementing `IEquatable`.